src/mbgl/math/log2.cpp

```diff
@@ -21,15 +21,9 @@ uint32_t ceil_log2(uint64_t x) {
     return y;
 }
 
-double log2(double x) {
-// log2() is producing wrong results on ARMv5 binaries
-// running on ARMv7+ CPUs.
-#if defined(__ANDROID__)
-    return std::log(x) / 0.6931471805599453; // log(x) / log(2)
-#else
-    return ::log2(x);
-#endif
-}
+template <> float log2(float x);
+template <> double log2(double x);
```

Comment:
Yeah, but they are only needed when you want to force certain instances to exist even if they're not used, or when the template definition is in an implementation file.